render-ws-spark-client/src/test/java/org/janelia/render/client/spark/pipeline/AlignmentPipelineParametersTest.java

```diff
@@ -0,0 +1,39 @@
+package org.janelia.render.client.spark.pipeline;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.janelia.render.client.newsolver.setup.AffineBlockSolverSetup;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests the {@link AlignmentPipelineParameters} class.
+ *
+ * @author Eric Trautman
+ */
+public class AlignmentPipelineParametersTest {
+
+    @Test
+    public void testJson()
+            throws IOException {
+        final AlignmentPipelineParameters pipelineParameters =
+                AlignmentPipelineParameters.fromJsonFile("src/test/resources/pipeline/msem_alignment_pipeline.json");
+
+        Assert.assertNotNull("deserialized parameters are null", pipelineParameters);
+
+        final AffineBlockSolverSetup affineBlockSolverSetup = pipelineParameters.getAffineBlockSolverSetup();
+        Assert.assertNotNull("affineBlockSolverSetup is null", affineBlockSolverSetup);
+
+        Assert.assertEquals("incorrect affineBlockSolverSetup.targetStack.stackSuffix value parsed",
+                            "_align_pipe_alt_aa", affineBlockSolverSetup.targetStack.stackSuffix);
+
+        Assert.assertTrue("incorrect affineBlockSolverSetup.targetStack.completeTargetStack value parsed",
+                          affineBlockSolverSetup.targetStack.completeStack);
+
+        final List<AlignmentPipelineStep> stepClients = pipelineParameters.buildStepClients();
+        Assert.assertNotNull("stepClients is null", stepClients);
+        Assert.assertEquals("incorrect number of stepClients", 7, stepClients.size());
```

Comment:
This test method does a lot of things:
- checking if JSON deserialization produces a non-null object;
- verify that the correct target stack settings are loaded;
- testing the conversion to `AlignmentPipelineStep`.
Should we maybe split this into several tests with more descriptive names? This way the test could also serve as kind of documentation.

Reply:
Yeah - I was being lazy.  I've split things up now.